### PR TITLE
feat(plugins): Use SpinnakerExtensionPoint for orca-api extension points, skipping Task for now

### PR DESCRIPTION
**Pull request by jonsie**
Depends on: https://github.com/spinnaker/kork/pull/632

This implementation uses `SpinnakerExtensionPoint` which is added in the above PR.

### CODE REVIEW

File: orca-core/src/main/java/com/netflix/spinnaker/orca/TaskResolver.java [hunk 1]
```diff
@@ -113,7 +113,8 @@ public Task getTask(@Nonnull Class<? extends Task> taskType) {
   @Nonnull
   public Class<? extends Task> getTaskClass(@Nonnull String taskTypeIdentifier) {
     try {
-      return getTask(taskTypeIdentifier).getClass();
+      Task task = getTask(taskTypeIdentifier);
+      return (Class<? extends Task>) task.getExtensionClass();
```
Comment (jonsie) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/TaskResolver.java:117]: When the extension is proxied, `TaskExecution.type` class was being serialized as a `Proxy` object.  This fixes that, and allows `TaskTypeDeserializer` to deserialize the proxied object to the correct Task type.

**Comment by jonsie** [comment]
When spinnaker/kork#632 merges and we cut a kork release, this test here will pass:

https://github.com/spinnaker/orca/blob/master/orca-plugins-test/src/test/kotlin/com/netflix/spinnaker/orca/plugins/test/OrcaPluginsTest.kt#L74

File: gradle.properties [hunk 2]
```diff
@@ -1,7 +1,7 @@
 enablePublishing=false
 fiatVersion=1.18.3
 keikoVersion=3.5.0
-korkVersion=7.41.1
+korkVersion=999.0.0-local-SNAPSHOT
```
Comment (srekapalli) [inline gradle.properties:4]: korkVersion seems odd .....?

Comment (cfieber) [reply gradle.properties:4]: this was built on spinnaker/kork#632

**Comment by jonsie** [comment]
Backed out the `TaskResolver` and `Task` changes here due to extensive unit test changes that are required.  I'll put up a separate PR for that work, this will at least un-block the Kork bump.

**Pull request merged**